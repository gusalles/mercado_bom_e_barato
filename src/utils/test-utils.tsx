import { render } from '@testing-library/react';
import { ReactElement } from 'react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

export * from '@testing-library/react';

export function renderWithClient(ui: ReactElement) {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  });

  return render(
    <QueryClientProvider client={queryClient}>{ui}</QueryClientProvider>
  );
}
