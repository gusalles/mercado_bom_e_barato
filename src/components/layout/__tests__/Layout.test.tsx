import { render, screen } from '@/utils/test-utils';
import { Layout } from '../index';

describe('Layout', () => {
  it('should render', () => {
    render(<Layout>Elemento filho</Layout>);

    const main = screen.getByRole('main');

    expect(main).toMatchSnapshot();
  });
});
