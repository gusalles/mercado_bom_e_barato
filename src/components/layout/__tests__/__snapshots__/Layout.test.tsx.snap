// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Layout > should render 1`] = `
<main
  class="css-16bb68u"
>
  Elemento filho
</main>
`;
